oauthproxy.go

```diff
@@ -245,6 +256,12 @@ func NewOAuthProxy(opts *Options, validator func(string) bool) *OAuthProxy {
 		refresh = fmt.Sprintf("after %s", opts.CookieRefresh)
 	}
 
+	whitelistDomains := opts.WhitelistDomains
+	if len(serveMuxes) >= 2 && len(whitelistDomains) == 0 && opts.CookieDomain != "" {
+		// by default the cookie-domain is allowed
+		whitelistDomains = append(whitelistDomains, opts.CookieDomain)
```

Comment:
If the cookie-domain is set to start with a `.`, then this would allow all subdomains of the cookie domain to be authenticated, is this the intention? Do we need to check the `.` prefix on the cookie domain before appending it to the whitelist?

Reply:
Yes, it's intentional, it's the default for when domain routing is enabled, in order to make it work. I couldn't think of any issue with it, is there an issue?

Reply:
If there's an issue, we can instead of setting this default require the user to set `whitelist-domains` to `.<cookie-domain>` when using subdomain routing. Would that be acceptable?

Reply:
I would prefer if this were a documented configuration rather than a default. Since it is a security consideration that users should be aware of.

Also, the user could set an explicit list of subdomains for routing, they may not necessarily want a wildcard even when using subdomain routing if I've understood correctly

Reply:
OK I removed that part and added a note in documentation, please check my new commit.
Now whenever subdomain routing is used, the whitelist must be set or nothing will work.

NB there is a difference in the naming between the `whitelist-domain` command-line flag and `whitelist_domains` config option (notice the `s`).